Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -1638,6 +1666,49 @@ namespace Multiplayer
         }
     }
 
+    void MultiplayerSystemComponent::MetricsEvent()
+    {
+        const auto& networkInterfaces = AZ::Interface<AzNetworking::INetworking>::Get()->GetNetworkInterfaces();
+        for (const auto& networkInterface : networkInterfaces)
+        {
+            if (networkInterface.second->GetType() != ProtocolType::Udp)
+            {
+                continue;
+            }
+            if (networkInterface.second->GetTrustZone() != TrustZone::ExternalClientToServer)
+            {
+                continue;
+            }
+
+            const NetworkInterfaceMetrics& metrics = networkInterface.second->GetMetrics();
+            SET_PERFORMANCE_STAT(MultiplayerStat_TotalTimeSpentUpdatingMs, metrics.m_updateTimeMs);            
+            SET_PERFORMANCE_STAT(MultiplayerStat_TotalSendTimeMs, metrics.m_sendTimeMs);
+            SET_PERFORMANCE_STAT(MultiplayerStat_TotalSentPackets, metrics.m_sendPackets);
+            SET_PERFORMANCE_STAT(MultiplayerStat_TotalSentBytesAfterCompression, metrics.m_sendBytes);
+            SET_PERFORMANCE_STAT(MultiplayerStat_TotalSentBytesBeforeCompression, metrics.m_sendBytesUncompressed);
+            SET_PERFORMANCE_STAT(MultiplayerStat_TotalResentPacketsDueToPacketLoss, metrics.m_resentPackets);
+            SET_PERFORMANCE_STAT(MultiplayerStat_TotalReceiveTimeInMs, metrics.m_recvTimeMs);
+            SET_PERFORMANCE_STAT(MultiplayerStat_TotalReceivedPackets, metrics.m_recvPackets);
+            SET_PERFORMANCE_STAT(MultiplayerStat_TotalReceivedBytesAfterCompression, metrics.m_recvBytes);
+            SET_PERFORMANCE_STAT(MultiplayerStat_TotalReceivedBytesBeforeCompression, metrics.m_recvBytesUncompressed);
+            SET_PERFORMANCE_STAT(MultiplayerStat_TotalPacketsDiscardedDueToLoad, metrics.m_discardedPackets);
+
+            break; // Assuming there is only one network interface for communicating with clients
+        }
+    }
+
+    void MultiplayerSystemComponent::OnPhysicsPresimulate([[maybe_unused]] float dt)
+    {
+        m_startPhysicsTickTime = AZStd::chrono::steady_clock::now();
+    }
+
+    void MultiplayerSystemComponent::OnPhysicsPostsimulate([[maybe_unused]] float dt)
+    {
+        const auto duration = AZStd::chrono::duration_cast<AZStd::chrono::microseconds>(
+            AZStd::chrono::steady_clock::now() - m_startPhysicsTickTime);
+        SET_PERFORMANCE_STAT(MultiplayerStat_PhysicsFrameTimeUs, AZ::TimeUs{ duration.count() });
+    }
```

Comment:
I'm noticing the code base seems to switch between Pre/PostSimulate and Pre/Postsimulate. It may be good to see if sig-physics has a preference here.

Reply:
Huh, well, this file: C:\git\o3de\Code\Framework\AzFramework\AzFramework\Physics\Common\PhysicsEvents.h has it as
```cpp
        //! Event triggers at the beginning of the SystemInterface::Simulate call.
        //! Parameter is the total time that the physics system will run for during the Simulate call.
        using OnPresimulateEvent = AZ::Event<float>;

        //! Event triggers at the end of the SystemInterface::Simulate call.
        //! Parameter is the total time that the physics system will run for during the Simulate call.
        using OnPostsimulateEvent = AZ::Event<float>;
```

Reply:
But other cases use `S`imulate, so I'll switch to that.